python/shotgun_desktop/startup.py

```diff
@@ -191,13 +192,22 @@ def __initialize_sgtk(sgtk, app_bootstrap):
         dm = sgtk.util.CoreDefaultsManager()
         sg_auth = shotgun_authentication.ShotgunAuthenticator(dm)
 
-        # get the current user
-        user = sg_auth.get_default_user()
-        logger.info("Setting current user: %r" % user)
-        sgtk.set_authenticated_user(user)
+        # get the current user for the pipeline.
```

Comment:
i feel there is no explanation here about core versions and why we need to do this - i think that could be helpful for future code readers.